authority/admin/api/handler.go

```diff
@@ -12,24 +12,26 @@ import (
 
 // Handler is the Admin API request handler.
 type Handler struct {
-	acmeResponder   ACMEAdminResponder
-	policyResponder PolicyAdminResponder
+	acmeResponder    ACMEAdminResponder
+	policyResponder  PolicyAdminResponder
+	webhookResponder WebhookAdminResponder
 }
 
 // Route traffic and implement the Router interface.
 //
 // Deprecated: use Route(r api.Router, acmeResponder ACMEAdminResponder, policyResponder PolicyAdminResponder)
 func (h *Handler) Route(r api.Router) {
-	Route(r, h.acmeResponder, h.policyResponder)
+	Route(r, h.acmeResponder, h.policyResponder, h.webhookResponder)
 }
 
 // NewHandler returns a new Authority Config Handler.
 //
 // Deprecated: use Route(r api.Router, acmeResponder ACMEAdminResponder, policyResponder PolicyAdminResponder)
-func NewHandler(auth adminAuthority, adminDB admin.DB, acmeDB acme.DB, acmeResponder ACMEAdminResponder, policyResponder PolicyAdminResponder) api.RouterHandler {
+func NewHandler(auth adminAuthority, adminDB admin.DB, acmeDB acme.DB, acmeResponder ACMEAdminResponder, policyResponder PolicyAdminResponder, webhookResponder WebhookAdminResponder) api.RouterHandler {
```

Comment:
I never liked adding the responders like this. Now that we have the need to change this, we might want to do:

```go
type HandlerOption func(*Handler)

func WithACMEResponder(r ACMEAdminResponder) HandlerOption {
    return func(*Handler) {
        h.acmeResponser = r
    }
}

func WithPolicyResponder(r PolicyAdminResponder) HandlerOption {
    return func(*Handler) {
        h.policyResponser = r
    }
}

func WithWebhookResponder(r WebhookAdminResponder) HandlerOption {
    return func(*Handler) {
        h.webhookResponser = r
    }
}

func NewHandler(auth adminAuthority, adminDB admin.DB, acmeDB acme.DB, opts ...HandlerOption) api.RouterHandler {
    h := new(Handler)
    for _, fn := range opts {
        fn(h)
    }
    return h
}
```

Reply:
This method is deprecated, and it should not be used. You should remove this change.

Reply:
Can I delete it?

Reply:
Some of them, not all, in this case, because this is our own admin you can.

Reply:
d48f6740